### PR TITLE
Feature/filacms 5 create abstract content resource

### PR DESCRIPTION
**Pull request by jeremy-portable**
- Implemented Abstract Content Resource
- Extended it as Page model
- Implemented Resource for the page model
- Implemented Tiptap Editor
- Added Revisionable to track revisions
- Added feature to revert specific revisions

### CODE REVIEW

File: src/Events/ContentCreating.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+class ContentCreating
+{
+    use Dispatchable, InteractsWithSockets, SerializesModels;
+ 
+    /**
+     * Create a new event instance.
+     */
+    public function __construct( public Page $page ) {
+        $page->created_user_id = auth()->user()->id;
+        $page->updated_user_id = auth()->user()->id;
```
Comment (kyoungportable) [inline src/Events/ContentCreating.php:20]: Can we make sure we're only doing this if the slug is blank?  The user should potentially be able to override the slug.

File: src/Filament/Resources/PageResource.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
+<?php
```
Comment (kyoungportable) [inline src/Filament/Resources/PageResource.php:1]: Suggest we make all of these `abstract`, e.g `src/Filament/Resources/AbstractContentResource.php` etc.  Then create `src/Filament/Resources/PageResource.php` extending `AbstractContentResource`, not just `Resource`.

File: database/migrations/2024_02_09_400000_create_contents_table.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
```
Comment (kyoungportable) [inline database/migrations/2024_02_09_400000_create_contents_table.php:13]: Table name here should be 'pages', as the concrete implementation is pages.

File: src/Commands/InstallCommand.php [hunk 4]
```diff
@@ -43,6 +43,10 @@ public function handle()
 
         $this->info('Adding permissions');
 
+        $this->info('Creating Custom Filament Theme');
+        $this->call('make:-filament theme');
```
Comment (kyoungportable) [inline src/Commands/InstallCommand.php:47]: Is the '-' meant to be there?

**Review by kyoungportable — CHANGES_REQUESTED** [review]

File: src/Commands/InstallCommand.php (hunk 4 above)
Comment (jeremy-portable) [reply src/Commands/InstallCommand.php:47]: Yup sorry

**Review by kyoungportable — APPROVED** [review]

**Pull request merged**